packages/wrangler/src/proxy.ts

```diff
@@ -80,6 +80,22 @@ export function usePreviewServer({
     { request: IncomingMessage; response: ServerResponse }[]
   >([]);
 
+  /**
+   * The session doesn't last forever, and will evetually drop
+   * (usually within 5-15 minutes). When that happens, we simply
+   * restart the effect, effectively restarting the server. We use
+   * a state sigil as an effect dependency to do so.
+   */
+  const [retryServerSetupSigil, setRetryServerSetupSigil] = useState<number>(0);
+  function retryServerSetup() {
+    // Let's log this message every 3 times we retry,
+    // so it's not too spammy.
+    if ((retryServerSetupSigil + 1) % 3 === 0) {
+      console.log("reconnecting to preview server...");
+    }
```

Comment:
If it is not important for the user to be aware of reconnecting 2 out of each 3 times, then do they need to be aware of it at all?

I would lean more towards all-or-nothing.

And then, if it really is too annoying, then putting it behind some kind of verbose flag.
Or even better making a kind of "status bar" in the `Dev` UI, so that this can pop up occasionally and then disappear without leaving behind a log of logs.